p2p/src/config.rs

```diff
@@ -15,6 +15,54 @@
 
 use serde::{Deserialize, Serialize};
 
+pub const MDNS_DEFAULT_QUERY_INTERVAL: u64 = 0;
+pub const MDNS_IPV6_DEFAULT_IPV6_STATE: bool = false;
+
+/// Multicast DNS configuration.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "state")]
+pub enum MdnsConfig {
+    Enabled {
+        /// Interval (in milliseconds) at which to poll the network for new peers.
+        query_interval: u64,
+        /// Use IPv6 for multicast DNS
+        enable_ipv6_mdns_discovery: bool,
+    },
+    Disabled,
+}
+
+impl MdnsConfig {
+    pub fn new() -> Self {
+        MdnsConfig::Disabled
+    }
+
+    pub fn from_options(
+        enable_mdns: bool,
+        query_interval: Option<u64>,
+        enable_ipv6_mdns_discovery: Option<bool>,
+    ) -> Self {
+        if enable_mdns {
+            MdnsConfig::Enabled {
+                query_interval: query_interval.unwrap_or(MDNS_DEFAULT_QUERY_INTERVAL),
```

Comment:
If mDNS is enabled the default query time is 0?

Reply:
Yes, it comes from libp2p